app/controllers/speakers_controller.rb

```diff
@@ -7,7 +7,12 @@ def destroy
     proposal = speaker.proposal
     speaker.destroy
 
-    flash[:info] = "You've withdrawn from this proposal."
-    redirect_to proposal_url(slug: proposal.event.slug, uuid: proposal)
+    if current_user.id == speaker.id
```

Comment:
Changed this because if a speaker withdrew him/herself, they were redirected back to the proposal page, which they no longer had access to, so it errored out.